### PR TITLE
Account confirmation, Login saving, Styling and Routing

### PR DESCRIPTION
**Pull request by aizon2456**
=> account confirmation, remember_me checkbox, styling and redirecting/authentication handling

Users must now confirm their accounts through email (see Issue #61), which forced the redirecting/routing performed in the ApplicationController to be revised. Styling of the header and flash messages have been corrected (the <p> notices should be addressed similarly in the future, see Issue #63), and the remember_me checkbox from Issue #58 is fixed now.

### CODE REVIEW

File: app/mailers/user_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class UserMailer < ActionMailer::Base
+  default :from => "ian.m.y.wong@gmail.com"
```
Comment (Rosengren) [inline app/mailers/user_mailer.rb]: Using your own email. Risky. We should use something like Heroku Postmark. Checkout the add-ons on heroku for email services

File: app/views/pages/confirm.html.erb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<div class="ui container aligned-text">
+In order to take advantage of the wonderful workings of TeamUp, it's important that we know that you're as committed to this process as we are. By confirming your email, you agree to sign away all doubts and worries to us, and let our super-duper algorithms do all the worrying for you.
```
Comment (Rosengren) [inline app/views/pages/confirm.html.erb:2]: love it

File: app/views/user_mailer/register_account.text.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Welcome to TeamUp, <%= @user.username %>!
+
+In order to activate your account and all the other schtuff we provide, we need you to click the link below to verify your account.
+
+If you would rather not TeamUp, feel free to ignore this!
+
+<%= confirm_account_user_url(@user.confirmation_action) %>
```
Comment (Rosengren) [inline app/views/user_mailer/register_account.text.erb:7]: Why is this part repeated in the register_account.html.erb? Is there any way to reuse the same text?

File: config/environments/development.rb [hunk 4]
```diff
@@ -14,7 +14,21 @@
...
+  }
+  # SMTP settings for gmail
+  config.action_mailer.smtp_settings = {
+    :address              => "smtp.gmail.com",
+    :port                 => 587,
+    :user_name            => ENV['gmail_username'],
+    :password             => ENV['gmail_password'],
+    :authentication       => "plain",
+    :enable_starttls_auto => true
+  }
```
Comment (Rosengren) [inline config/environments/development.rb:31]: Will emails work in production? I see you only edited the development config file.

File: .gitignore [hunk 5]
```diff
@@ -17,3 +17,6 @@
 /tmp
 .DS_Store
 *~
+
+# Ignore application configuration
+/config/application.yml
```
Comment (Rosengren) [inline .gitignore:22]: If I understand this correctly, this YAML file is created/used by the figaro gem. I've never used figaro before but the description says its a Heroku-friendly config file. I'm assuming we will need some sort of application file for the heroku (production) version

**Comment by Rosengren** [comment]
Looks good! See comments before merging

File: app/mailers/user_mailer.rb (hunk 1 above)
Comment (aizon2456) [inline app/mailers/user_mailer.rb]: Created a Gmail account for this purpose, see upcoming commit for change.

File: app/views/user_mailer/register_account.text.erb (hunk 3 above)
Comment (aizon2456) [inline app/views/user_mailer/register_account.text.erb:7]: We can investigate the potential creation of a template for handling text vs. html, but according to the Rails documentation and Google, without an additional gem or third-party piece of software, it cannot be done. Opened #66 to address this.

File: config/environments/development.rb (hunk 4 above)
Comment (aizon2456) [inline config/environments/development.rb:31]: Correct, they will not work. However, all we will need is for the application.yml to be configured for production. TBD.

File: .gitignore (hunk 5 above)
Comment (aizon2456) [inline .gitignore:22]: Correct, we'll need to handle this as Issue #62 indicates.

**Comment by aizon2456** [comment]
Newest commit fixes aforementioned issues, and handles an issue with the account_confirmed? method not being found.

**Pull request merged**